packages/amplication-server/src/core/subscription/subscription.service.ts

```diff
@@ -230,59 +114,47 @@ export class SubscriptionService {
   async handleUpdateSubscriptionStatusEvent(
     updateStatusDto: UpdateStatusDto
   ): Promise<void> {
+    const data =
+      this.mapUpdateStatusDtoToUpsertSubscriptionInput(updateStatusDto);
+
     switch (updateStatusDto.type) {
-      case "subscription.created": {
-        const createSubscriptionInput =
-          this.mapUpdateStatusDtoToCreateSubscriptionInput(updateStatusDto);
-        const subs = await this.getSubscriptionById(updateStatusDto.id);
-        if (subs) {
-          await this.create(updateStatusDto.id, createSubscriptionInput);
-          await this.trackUpgradeCompletedEvent(
-            createSubscriptionInput.plan,
-            updateStatusDto.metadata?.userId,
-            updateStatusDto.customer.id
-          );
-        } else {
-          this.logger.warn(
-            `Trying to create an existing subscription with id: ${updateStatusDto.id}`,
-            updateStatusDto
-          );
-        }
-        break;
-      }
+      case "subscription.created":
       case "subscription.updated":
       case "subscription.expired":
       case "subscription.canceled": {
-        const updateSubscriptionInput =
-          this.mapUpdateStatusDtoToUpdateSubscriptionInput(updateStatusDto);
-        const subs = await this.getSubscriptionById(updateStatusDto.id);
-        if (subs) {
-          await this.update(updateStatusDto.id, updateSubscriptionInput);
-        } else {
-          this.logger.warn(
-            `Trying to update a non-existing subscription with id: ${updateStatusDto.id}`,
-            updateStatusDto
-          );
-        }
+        await this.prisma.subscription.upsert({
+          where: {
+            id: updateStatusDto.id,
+          },
+          create: {
+            id: updateStatusDto.id,
+            workspaceId: data.workspaceId,
+            status: data.status,
+            subscriptionPlan: data.plan,
+            subscriptionData:
+              data.subscriptionData as unknown as Prisma.InputJsonValue,
+          },
+          update: {
+            status: data.status,
+            subscriptionData:
+              data.subscriptionData as unknown as Prisma.InputJsonValue,
+          },
+        });
         break;
       }
     }
+    if (updateStatusDto.type === "subscription.created") {
```

Comment:
If Stigg sends multiple `subscription.created` events for the same subscription, we might end up sending this event multiple times to BI